lib/Sema/SemaDecl.cpp

```diff
@@ -8521,16 +8526,20 @@ Sema::ActOnFunctionDeclarator(Scope *S, Declarator &D, DeclContext *DC,
     }
   }
 
-  // Checked C - checked function return type checking
+  // Checked C - type restriction on declarations in checked block
+  // function parameter & return type is not allowed to use unchecked type
+  // in checked blocks
   if (D.getDeclSpec().isCheckedSpecified()) {
-    // current scope - function definition/body scope, checked propery is from
-    // checked function or checked scope
-    // Checked C - in checked function, check return/param type
-    // In checked scope, no prototype function is not allowed to use
-    DiagnoseCheckedDecl(NewFD);
+    // current scope - function definition/body scope
+    // checked propery is from checked function or checked scope keyword
```

Comment:
Typo: propery->property.   Also please add some periods here.  
I think there should be periods here:
// Current scope checked property is from checked function or
// checked scope keyword.
// In checked functions, check function/param types.
// In checked scopes, no prototypes functions are not allowed.